src/test/java/io/cdap/plugin/ariba/source/AribaServicesTest.java

```diff
@@ -0,0 +1,746 @@
+    "        },\n" +
+    "        {\n" +
+    "          \"type\": [\n" +
+    "            \"array\",\n" +
+    "            \"null\"\n" +
+    "          ],\n" +
+    "          \"properties\": {}\n" +
+    "        }\n" +
+    "      ]\n" +
+    "        }\n" +
+    "        \n" +
+    "      ]\n" +
+    "    }\n" +
+    "  }\n" +
+    "}";
+  private AribaServices aribaServices;
+  private AribaPluginConfig pluginConfig;
+  private MockPipelineConfigurer pipelineConfigurer;
+
+  @Before
+  public void setUp() {
+    pipelineConfigurer = new MockPipelineConfigurer(null);
+    pluginConfig = new AribaPluginConfig("unit-test-ref-name", "https://openapi.au.cloud.ariba.com",
+                                         "prod", "CloudsufiDSAPP-T",
+                                         "SourcingProjectFactSystemView", "08ee0299-4849-42a4-8464-3abed75fc74e",
+                                         "c3B5wvrEsjKucFGlGkkhKSWUDqDRGE2Wds", "xryi0757SU8pEyk7ePc7grc7vgDXdz8O",
+                                         "2022-01-28T10:05:02Z", "2022-01-31T10:05:02Z");
+  }
+
+  @Test
+  public void testGetAccessToken() throws AribaException, IOException {
+    aribaServices = new AribaServices(pluginConfig);
+
+    String tokenUrl = String.format("https://%s", "https://api.au.cloud.ariba.com/v2/oauth/token");
+    HttpUrl.Builder builder = Objects.requireNonNull(HttpUrl.parse(tokenUrl))
+      .newBuilder().addPathSegments("TOKEN_PATH");
+
+    InputStream inputStream = new ByteArrayInputStream(jsonNode.getBytes(StandardCharsets.UTF_8));
+    new Expectations(AribaServices.class) {
+      {
+        aribaServices.generateTokenURL();
+        result = any;
+        minTimes = 0;
+
+        aribaServices.callAribaForToken((URL) any);
+        result = response;
+        minTimes = 0;
+
+        response.getResponseBody();
+        result = inputStream;
+        minTimes = 0;
+
+        response.getHttpStatusCode();
+        result = 200;
+      }
+    };
+
+    String accessToken = aribaServices.getAccessToken();
+    Assert.assertNotNull(accessToken);
+    Assert.assertEquals(accessToken, "jiuokiopu");
+
+  }
+
+  @Test
+  public void testGenerateTokenURL() throws AribaException {
+    aribaServices = new AribaServices(pluginConfig);
+    Assert.assertNotNull(aribaServices.generateTokenURL());
+  }
+
+  @Test
+  public void testCallAribaForTokenForError() throws IOException {
+    aribaServices = new AribaServices(pluginConfig);
+    String tokenUrl = String.format("https://%s", "https://api.au.cloud.ariba.com/v2/oauth/token");
+    HttpUrl.Builder builder = Objects.requireNonNull(HttpUrl.parse(tokenUrl))
+      .newBuilder().addPathSegments("TOKEN_PATH");
+    new Expectations(AribaServices.class) {
+      {
+        aribaServices.httpAribaTokenCall(builder.build().url());
+        result = new IOException();
+        minTimes = 0;
+      }
+    };
+    try {
+      aribaServices.callAribaForToken(builder.build().url());
```

Comment:
fail tests if expected exception is not thrown

Reply:
Fixed here (https://github.com/data-integrations/ariba-plugins/pull/2/commits/9e5462f14445ea32445eb3bdd766d3d83f76499b)